burp/BurpExtender.java

```diff
@@ -161,122 +129,30 @@ public IHttpService getHttpService() {
+        if (columnIndex == 0) {
+//        String s = new String(logEntry.requestResponse.getRequest(), StandardCharsets.UTF_8);
+//        return s; //callbacks.getToolName(logEntry.tool);
+            return logEntry;
+        } else {
+            return "";
         }
     }
 
-        //
-// extend JTable to handle cell selection
-//
-
-    // Table for 2nd log set to use
-    public class MyTableModel extends AbstractTableModel {
-
-        //
-        // extend AbstractTableModel
-        //
-
-        @Override
-        public int getRowCount() { return log2.size(); }
-
-        @Override
-        public int getColumnCount() {
-            return 1;
-        }
-
-        @Override
-        public String getColumnName(int columnIndex) {
-            switch (columnIndex) {
-                case 0:
-                    return "HTTP Response";
-//            case 1:
-//                return "URL";
-                default:
-                    return "";
-            }
-        }
-
-        @Override
-        public Class<?> getColumnClass(int columnIndex) {
-            return String.class;
-        }
-
-        @Override
-        public Object getValueAt(int rowIndex, int columnIndex) {
-            // Adele note: ignoring rowIndex for now
-            LogEntry logEntry = log2.get(rowIndex);
-
-            switch (columnIndex) {
-                case 0:
-                    String s = new String(logEntry.requestResponse.getResponse(), StandardCharsets.UTF_8);
-                    return s; //callbacks.getToolName(logEntry.tool);
-                case 1:
-                    return logEntry.url.toString();
-                default:
-                    return "";
-
-                //return log2.get(rowIndex);
-
-            }
-        }
+    private record LogEntry(int tool, IHttpRequestResponsePersisted requestResponse, URL url) {
     }
 
-    private class Table extends JTable {
-        public String tableName;
-        public List<LogEntry> table_log;
-
-        public Table(TableModel tableModel, String tableName, List<LogEntry> log) {
-            super(tableModel);
-            this.tableName = tableName;
-            this.table_log = log;
-
-        }
-
-        @Override
-        public void changeSelection(int row, int col, boolean toggle, boolean extend) {
-            // show the log entry for the selected row
-            LogEntry logEntry = this.table_log.get(row);
-            requestViewer.setMessage(logEntry.requestResponse.getRequest(), true);
-            responseViewer.setMessage(logEntry.requestResponse.getResponse(), false);
-            currentlyDisplayedItem = logEntry.requestResponse;
-
-            super.changeSelection(row, col, toggle, extend);
-        }
+    // Don't do anything with these. They're artifacts from a necessary (currently) thing we can't get rid of.
+    @Override
+    public int getRowCount() {
+        return 0;
     }
 
```

Comment:
Is this still a needed artifact?

Reply:
Yes. It's part of the AbstractTableModel that is used, and because we use the fireTableRowsInserted currently, we need to keep that around. I haven't seen a reason for it to be there yet, but we can't remove it without causing compile errors.